scripts/create_deprecated_modules.lean

```diff
@@ -0,0 +1,291 @@
+-/
+def getHeader (fname fileContent : String) (keepTrailing : Bool) : IO String := do
+  let (stx, _) ← Parser.parseHeader (Parser.mkInputContext fileContent fname)
+  let stx := if keepTrailing then stx.raw else stx.raw.unsetTrailing
+  let some substring := stx.getSubstring? | throw <| .userError "No substring: we have a problem!"
+  let upToAllImports : Substring := {substring with startPos := 0}
+  return upToAllImports.toString
+
+/--
+`getHeaderFromFileName fname keepTrailing` is similar to `getHeader`, except that it assumes that
+`fname` is the actual path of a file, and uses `fname`'s content as input to `getHeader`.
+-/
+def getHeaderFromFileName (fname : String) (keepTrailing : Bool) : IO String := do
+  getHeader fname (← IO.FS.readFile fname) keepTrailing
+
+/--
+`mkDeprecation customMessage` returns the formatted syntax
+`deprecated_module "customMessage" (since := "YYYY-MM-DD")`,
+where the date is today's date.
+-/
+def mkDeprecation (customMessage : String := "Auto-generated deprecation") :
+    CommandElabM Format := do
+  let msgStx := if customMessage.isEmpty then none else some <| Syntax.mkStrLit customMessage
+  -- Get the current date in UTC: we don't want this to depend on the user computer's time zone.
+  let dateStx := Syntax.mkStrLit s!"{(← Std.Time.DateTime.now (tz := .UTC)).toPlainDate}"
+  let stx ← `(command|deprecated_module $[$msgStx]? (since := $dateStx))
+  liftCoreM <| PrettyPrinter.ppCategory `command stx
+
+/--
+The command `#create_deprecated_module filePath (comment)? (write)?` generates a module deprecation.
+
+Writing
+```lean
+#create_deprecated_module path/To/DeletedFile.lean "This file is no longer relevant"
+```
+checks that `path/To/DeletedFile.lean` is not currently present, but was present in `Mathlib/`
+at some point.
+
+If the check is successful, then it reports on its findings, shows how the corresponding
+deprecated module should look like, using `"This file is no longer relevant"` as the (optional)
+comment.
+
+If the message is not explicitly used, `#create_deprecated_module` defaults to
+`"Auto-generated deprecation"`.
+If you wish there to be no comment, use `#create_deprecated_module path/To/DeletedFile.lean ""`.
+
+Finally, if everything looks correct, adding a final `write` actually generates the file:
+```lean
+#create_deprecated_module path/To/DeletedFile.lean "This file is no longer relevant" write
+```
+-/
+syntax "#create_deprecated_module " str (ppSpace str)? (&" write")? ppLine :
+  command
+
+/-- `processPrettyOneLine log msg` takes as input two strings `log` and `msg`.
+It expects `log` to be a line in the output of `git log --pretty=oneline`:
+it should look like `<hash> <PRdescr>`.
+
+It also expects `log` to be either `last modified` or `deleted` and
+it returns the pair `(<hash>, <msg> in <PRdescr> <hash>)`, formatted as a collapsible message.
+-/
+def processPrettyOneLine (log msg : String) : String × MessageData :=
```

Comment:
I've just tried to read this in detail, and something is off. The doc-comment says two different things, and does not match what the function does. Can you clarify what you mean? Thanks!

Reply:
Well spotted!  The second `log` should have been `msg`!